test/service/tests/integration_test.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use cumulus_primitives::ParaId;
+use futures::join;
+use polkadot_primitives::v0::{Info, Scheduling};
+use polkadot_runtime_common::registrar;
+use sc_service::TaskExecutor;
+use substrate_test_runtime_client::AccountKeyring::*;
+
+#[substrate_test_utils::test]
+#[ignore]
+async fn integration_test(task_executor: TaskExecutor) {
+	let para_id = ParaId::from(100);
+
+	// start alice
+	let alice = polkadot_test_service::run_test_node(task_executor.clone(), Alice, || {}, vec![]);
+
+	// start bob
+	let bob = polkadot_test_service::run_test_node(
+		task_executor.clone(),
+		Bob,
+		|| {},
+		vec![alice.addr.clone()],
+	);
+
+	// ensure alice and bob can produce blocks
+	join!(alice.wait_for_blocks(2), bob.wait_for_blocks(2));
+
+	// call function to register the parachain
+	let function = polkadot_test_runtime::Call::Sudo(pallet_sudo::Call::sudo(Box::new(
```

Comment:
That should clearly be a default feature of the testing framework. I don't want to repeat this code over and over.

Reply:
https://github.com/paritytech/polkadot/pull/1733

Reply:
done in ce34da6